internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,189 @@
+package olm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	defaultIndexImage = "quay.io/operator-framework/upstream-opm-builder:latest"
+	defaultGRPCPort   = 50051
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	BundleAddMode string
+	BundleImage   string
+	DBPath        string
+	IndexImage    string
+	Namespace     string
+	Kubeclient    kubernetes.Interface
+
+	grpcPort int32
+}
+
+// Create returns a bundle registry pod built from an index image
+func (rp *RegistryPod) Create(ctx context.Context) (*corev1.Pod, error) {
+	// Make the pod spec
+	pod, err := rp.build()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod: %v", err)
+	}
+
+	// Check if Pod already exists
+	tmpPod, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Create pod in kubernetes cluster using the above pod definition and clientset
+			pod, err = rp.Kubeclient.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("error creating registry pod: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting existing registry pod: %v", err)
+		}
+	} else {
+		return tmpPod, nil
+	}
+
+	// Upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, fmt.Errorf("error getting pod %s: %w", pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	err = wait.PollImmediateUntil(time.Duration(1*time.Second), podCheck, ctx.Done())
+	if err != nil {
+		return nil, fmt.Errorf("error waiting for registry pod %s to run: %v", pod.Name, err)
+	}
+
+	//TODO: make pod logs available, maybe return log output to caller
+	_, err = rp.fetchPodLogs(ctx, pod)
+	if err != nil {
+		return pod, fmt.Errorf("error in fetching pod logs: %v", err)
+	}
+
```

Comment:
Why do we need to fetch the pod logs?

Reply:
@camilamacedo86 for `run bundle` and `run bundle-upgrade` commands, we talked about capturing the pod logs for debugging in the case of pod failures. For now, I'm using it in Create() function as a placeholder but it will be moved to appropriate function when the CLI is plugged in.